### PR TITLE
feat(go extractor): allow specifying go +build tags

### PR DESCRIPTION
**Pull request by justbuchanan**
Some go files are conditionally included in the build according to semantic comments at the top of the file. The new `--buildtags` flag allows specifying which of these tags/constraints to enable so that the right files get included by the extractor.

https://golang.org/pkg/go/build/#hdr-Build_Constraints

TODO: is there a cleaner way to have multi-value flags in go?

### CODE REVIEW

File: kythe/go/extractors/cmd/gotool/gotool.go [hunk 1]
```diff
@@ -50,6 +50,20 @@ var (
...
+
+func (af *arrayFlags) String() string {
+	return strings.Join(*af.array, ",")
+}
+
+func (af *arrayFlags) Set(value string) error {
+	*af.array = append(*af.array, value)
+	return nil
+}
+
```
Comment (jaysachs) [inline kythe/go/extractors/cmd/gotool/gotool.go]: I think, based on my recent, change, you'll also need a Get implementation.

Also, maybe "stringArrayFlag" as the type name?

File: kythe/go/extractors/cmd/gotool/gotool.go [hunk 2]
```diff
@@ -70,6 +84,7 @@ Options:
 	flag.StringVar(&bc.GOROOT, "goroot", bc.GOROOT, "Go system root")
 	flag.BoolVar(&bc.CgoEnabled, "gocgo", bc.CgoEnabled, "Whether to allow cgo")
 	flag.StringVar(&bc.Compiler, "gocompiler", bc.Compiler, "Which Go compiler to use")
+	flag.Var(&arrayFlags{array: &bc.BuildTags}, "buildtag", "Go +build tags to enable during extraction. May be specified multiple times.")
```
Comment (jaysachs) [inline kythe/go/extractors/cmd/gotool/gotool.go]: "buildtags"?

Also, what about plumbing them through the build?

File: kythe/go/extractors/cmd/gotool/gotool.go [hunk 3]
```diff
@@ -70,6 +84,7 @@ Options:
...
```
Comment (shahms) [inline kythe/go/extractors/cmd/gotool/gotool.go]: `flagutil.StringSet` from `kythe.io/kythe/go/util/flagutil`

Comment (shahms) [reply kythe/go/extractors/cmd/gotool/gotool.go]: or flagutil.StringList

Comment (justbuchanan) [reply kythe/go/extractors/cmd/gotool/gotool.go]: Awesome :+1:

File: kythe/go/extractors/cmd/gotool/gotool.go (hunk 2 above)
Comment (justbuchanan) [reply kythe/go/extractors/cmd/gotool/gotool.go]: Renamed to "buildtags".

What do you mean by "plumbing them through the build"? The extractor does already handle these correctly, it just didn't have a way to specify them on the command line.

Comment (creachadair) [reply kythe/go/extractors/cmd/gotool/gotool.go]: This extractor doesn't actually execute a build—it only simulates one. When running `go build` explicitly one includes tags in the same way, via a flag.

When building in Bazel, tags are handled differently: Bazel permits mixing packages in a single directory, so the the rules have to do some shenanigans to virtualize the directory structure and I believe file selection is done there.

**Pull request merged**